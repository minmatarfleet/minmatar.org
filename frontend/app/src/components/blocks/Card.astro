---
interface Props {
    [propName: string]: any;
}

const {
    ...attributes
} = Astro.props

import Flexblock from "@components/compositions/Flexblock.astro";
import Wrapper from "@components/compositions/Wrapper.astro";

import ComponentBlock from '@components/blocks/ComponentBlock.astro';
---

<ComponentBlock class="[ card card-animated ]" padding_block='0' padding_inline='0' {...attributes}>
    <Flexblock gap="0">
        <slot name="cover" />

        <Wrapper class="[ card-body ]" padding_block="var(--space-2xs)" padding_inline="var(--space-2xs)">
            <slot />
        </Wrapper>

        <Wrapper
            class="[ card-header ]"
            padding_block="var(--space-2xs)"
            padding_inline="var(--space-2xs)"
        >
            <slot name="header" />
        </Wrapper>

        <Wrapper class="[ card-actions ]" padding_block="var(--space-2xs)" padding_inline="var(--space-2xs)">
            <slot name="actions" />
        </Wrapper>
    </Flexblock>
</ComponentBlock>

<style lang="scss">
    .card {
        position: relative;
        max-height: 60vh;
        overflow: hidden;

        &:hover {
            .card-actions {
                opacity: 1;
            }
        }

        .card-body {
            position: absolute;
            bottom: 0;
            left: 0;
            right: 0;
        }

        .card-header {
            position: absolute;
            top: var(--space-3xs);
            left: var(--space-3xs);
        }

        .card-actions {
            position: absolute;
            top: 0;
            right: 0;
            opacity: 0;
        }
    }
</style>