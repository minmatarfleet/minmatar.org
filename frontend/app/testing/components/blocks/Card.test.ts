import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test } from "vitest";
import Card from "@components/blocks/Card.astro";

test("Card defaults", async () => {
  const container = await AstroContainer.create();
  const result = await container.renderToString(Card, {});

  expect(result).toMatchSnapshot();
});
