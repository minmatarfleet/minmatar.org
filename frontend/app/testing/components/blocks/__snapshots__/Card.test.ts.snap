// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card defaults 1`] = `"<div class="wrapper [ card component ] [ card card-animated ] light-transparency" data-astro-cid-idiuzidr="true" data-mod data-astro-cid-yv35cmqx style="--wrapper_padding_inline: 0;--wrapper_padding_inline_end: 0;--wrapper_padding_block: 0;--wrapper_max_width: 100%;">   <div class="flex-block gap-0" data-astro-cid-idiuzidr="true" data-mod style="--flex_block_gap: 0;">   <div class="wrapper [ card-body ]" data-astro-cid-idiuzidr="true" data-mod data-astro-cid-yv35cmqx style="--wrapper_padding_inline: var(--space-2xs);--wrapper_padding_inline_end: var(--space-2xs);--wrapper_padding_block: var(--space-2xs);--wrapper_max_width: 80rem;">    </div>   <div class="wrapper [ card-header ]" data-astro-cid-idiuzidr="true" data-mod data-astro-cid-yv35cmqx style="--wrapper_padding_inline: var(--space-2xs);--wrapper_padding_inline_end: var(--space-2xs);--wrapper_padding_block: var(--space-2xs);--wrapper_max_width: 80rem;">    </div>   <div class="wrapper [ card-actions ]" data-astro-cid-idiuzidr="true" data-mod data-astro-cid-yv35cmqx style="--wrapper_padding_inline: var(--space-2xs);--wrapper_padding_inline_end: var(--space-2xs);--wrapper_padding_block: var(--space-2xs);--wrapper_max_width: 80rem;">    </div>    </div>    </div>   "`;
